docker/test_env/nethermind.go

```diff
@@ -0,0 +1,203 @@
+	return g
+}
+
+func (g *Nethermind) StartContainer() (blockchain.EVMNetwork, error) {
+	r, err := g.getContainerRequest(g.Networks)
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+
+	l := logging.GetTestContainersGoTestLogger(g.t)
+	ct, err := docker.StartContainerWithRetry(g.l, tc.GenericContainerRequest{
+		ContainerRequest: *r,
+		Reuse:            true,
+		Started:          true,
+		Logger:           l,
+	})
+	if err != nil {
+		return blockchain.EVMNetwork{}, errors.Wrapf(err, "cannot start nethermind container")
+	}
+
+	host, err := GetHost(context.Background(), ct)
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	httpPort, err := ct.MappedPort(context.Background(), NatPort(TX_GETH_HTTP_PORT))
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	wsPort, err := ct.MappedPort(context.Background(), NatPort(TX_GETH_WS_PORT))
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+	executionPort, err := ct.MappedPort(context.Background(), NatPort(ETH2_EXECUTION_PORT))
+	if err != nil {
+		return blockchain.EVMNetwork{}, err
+	}
+
+	g.Container = ct
+	g.ExternalHttpUrl = FormatHttpUrl(host, httpPort.Port())
+	g.InternalHttpUrl = FormatHttpUrl(g.ContainerName, TX_GETH_HTTP_PORT)
+	g.ExternalWsUrl = FormatWsUrl(host, wsPort.Port())
+	g.InternalWsUrl = FormatWsUrl(g.ContainerName, TX_GETH_WS_PORT)
+	g.InternalExecutionURL = FormatHttpUrl(g.ContainerName, ETH2_EXECUTION_PORT)
+	g.ExternalExecutionURL = FormatHttpUrl(host, executionPort.Port())
+
+	networkConfig := blockchain.SimulatedEVMNetwork
+	networkConfig.Name = fmt.Sprintf("Simulated Eth2 (Nethermind %s)", g.consensusLayer)
+	networkConfig.URLs = []string{g.ExternalWsUrl}
+	networkConfig.HTTPURLs = []string{g.ExternalHttpUrl}
+
+	g.l.Info().Str("containerName", g.ContainerName).
+		Msg("Started Nethermind container")
+
+	return networkConfig, nil
+}
+
+func (g *Nethermind) GetInternalExecutionURL() string {
+	return g.InternalExecutionURL
+}
+
+func (g *Nethermind) GetExternalExecutionURL() string {
+	return g.ExternalExecutionURL
+}
+
+func (g *Nethermind) GetInternalHttpUrl() string {
+	return g.InternalHttpUrl
+}
+
+func (g *Nethermind) GetInternalWsUrl() string {
+	return g.InternalWsUrl
+}
+
+func (g *Nethermind) GetExternalHttpUrl() string {
+	return g.ExternalHttpUrl
+}
+
+func (g *Nethermind) GetExternalWsUrl() string {
+	return g.ExternalWsUrl
+}
+
+func (g *Nethermind) GetContainerName() string {
+	return g.ContainerName
+}
+
+func (g *Nethermind) GetContainer() *tc.Container {
+	return &g.Container
+}
+
+func (g *Nethermind) getContainerRequest(networks []string) (*tc.ContainerRequest, error) {
+	return &tc.ContainerRequest{
+		Name:            g.ContainerName,
+		Image:           fmt.Sprintf("nethermind/nethermind:%s", NETHERMIND_IMAGE_TAG),
+		Networks:        networks,
+		AlwaysPullImage: true,
+		// ImagePlatform: "linux/x86_64", // this breaks everything, don't try it
```

Comment:
So, can it work on x86_64?

Reply:
Someone who has one should try, I only have Apple Silicon. But they do publish arm64 and x86 image versions (unlike other clients), so I imagine it should work.